src/utils/filter-utils.js

```diff
@@ -309,7 +375,7 @@ export function getFieldDomain(data, field) {
  * @param {Object} dataset to perform the filter on
  * @param {Object[]} filters list of filters to use against dataset
  */
-export function filterData(dataset, filters) {
+export function filterData(dataset, filters, layers) {
```

Comment:
in this function, we can create a `filterFunctors(field, dataset.id, filter, layers)` instead of a `filtersToFields` with all the pre compyting before passing data point to it.  filterFunction is a function that takes `(data, index) => true || false`, it shouldn't does any computation rather than get the value from data and compare it to filter.value. which means by passing `field, dataset.id, filter, layers` to `getFilterFunction` we on longer need to pass it to `isDataMatchFilter`

implementation as below:

```js
  const appliedFilters = filters.filter(d => shouldApplyFilter(d, dataset.id));

  // Map filter against current dataset field
  const filterFunctors = appliedFilters.reduce((acc, filter) => {
    const fieldIndex = getDatasetFieldIndexForFilter(dataset, filter);
    const field = fieldIndex !== -1 ? fields[fieldIndex] : null;

    return {
      ...acc,
      [filter.id]: getFilterFunction(field, dataset.id, filter, layers)
    }
  }, {});


function getFilterFunction(field, dataId, filter, layers) {
  // field could be null
  let valueAccessor = data => field ? data[field.tableFieldIndex - 1] : null;

  switch (filter.type) {
    case FILTER_TYPES.range:
      return data => {
        return isInRange(valueAccessor(data), filter.value)
      };
    case FILTER_TYPES.multiSelect:
      return data => filter.value.includes(valueAccessor(data));
    case FILTER_TYPES.select:
      return data => valueAccessor(data) === filter.value;
    case FILTER_TYPES.timeRange:
      if (field && field.filterProp && Array.isArray(field.filterProp.mappedValue)) {
        valueAccessor = (data, index) => field.filterProp.mappedValue[index];
      } else {
        valueAccessor = data => timeToUnixMilli(valueAccessor(data), field.format);
      }
      return (data, index) => isInRange(valueAccessor(data, index), filter.value);
    case FILTER_TYPES.polygon:
      if (!(layers || layers.length === 0)) {
        return data => true;
      }

      const layerFilterFunctions = filter.layerId
        .map(id => layers.find(l => l.id === id))
        .filter(l => l && l.config.dataId === dataId)
        .map(layer => getPolygonFilterFunctor(layer, filter));

      return data => layerFilterFunctions.every(filterFunc => filterFunc(data));
    default:
      return data => true;
  }
};
```

Reply:
this looks way better, thank you

Reply:
had to change few things in the above code because it was causing the following error:
```
not ok 1 RangeError: Maximum call stack size exceeded
  ---
    operator: error
    expected: |-
      undefined
    actual: |-
      [RangeError: Maximum call stack size exceeded]
    at: Test.die (/Users/gmacri/Documents/projects/kepler.gl/node_modules/tape-catch/index.js:56:10)
    stack: |-
      RangeError: Maximum call stack size exceeded
          at valueAccessor (/Users/gmacri/Documents/projects/kepler.gl/src/utils/filter-utils.js:418:25)
          at valueAccessor (/Users/gmacri/Documents/projects/kepler.gl/src/utils/filter-utils.js:418:49)
```
@ line: ```valueAccessor = data => timeToUnixMilli(valueAccessor(data), field.format);```

The overall result is way better